test/metabase/upload_test.clj

```diff
@@ -1751,3 +1752,45 @@
                       (rows-for-table table)))
 
               (io/delete-file file))))))))
+
+(defn- upload-table-exists? [table]
+  ;; we don't need to worry about sql injection here
+  (-> (format "SELECT 1 FROM information_schema.tables WHERE table_name = '%s'" (:name table))
+      ((fn [sql] {:database (:db_id table), :type :native, :native {:query sql}}))
+      qp/process-query
+      :row_count
+      pos?))
+
+(deftest delete-upload!-test
```

Comment:
why do we keep this test here but not `metabase-enterprise.uploads.api-test`?

feel more stream lined to test the archived cards path there.